### DISCUSSION TITLE
The data alignment of ADC result

### DISCUSSION THREAD

**Discussion by tonny1983** [tock/tock#3651] [Q&A] [answered]
The `Client` will return a `u16` type for ADC result. It is without confusion for 16-bit ADC, but for some others, such as 12-bit or 8-bit ADC.
The problem is just as the title said, what is the data alignment of the `u16` number?
In practice, the result is **left alignment**, at least in the Raspberry Pi Pico board, which has just 12-bit ADC.
Are there definitions or rules in tock for the question?

BTW: As I know, the data alignment od ADC can be set in SMT32f serials.

**Comment by ppannuto** [comment, marked as answer]
https://docs.tockos.org/kernel/hil/adc/trait.adc#tymethod.sample

> All ADC samples will be the raw ADC value left-justified in the u16.

I believe this is what you are asking about?
